auth/src/main/java/com/firebase/ui/auth/ui/phone/CheckPhoneNumberFragment.java

```diff
@@ -0,0 +1,197 @@
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        mHandler = ViewModelProviders.of(getActivity()).get(PhoneNumberVerificationHandler.class);
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater,
+                             @Nullable ViewGroup container,
+                             @Nullable Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.fui_phone_layout, container, false);
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        mCountryListSpinner = view.findViewById(R.id.country_list);
+        mPhoneInputLayout = view.findViewById(R.id.phone_layout);
+        mPhoneEditText = view.findViewById(R.id.phone_number);
+        mSmsTermsText = view.findViewById(R.id.send_sms_tos);
+
+        mSmsTermsText.setText(getString(R.string.fui_sms_terms_of_service,
+                getString(R.string.fui_verify_phone_number)));
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O && getFlowParams().enableHints) {
+            mPhoneEditText.setImportantForAutofill(View.IMPORTANT_FOR_AUTOFILL_NO);
+        }
+        getActivity().setTitle(getString(R.string.fui_verify_phone_number_title));
+
+        ImeHelper.setImeOnDoneListener(mPhoneEditText, new ImeHelper.DonePressedListener() {
+            @Override
+            public void onDonePressed() {
+                onNext();
+            }
+        });
+        view.findViewById(R.id.send_code).setOnClickListener(this);
+        mCountryListSpinner.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                mPhoneInputLayout.setError(null);
+            }
+        });
+    }
+
+    @Override
+    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+        if (savedInstanceState != null || mCalled) { return; }
+        // Fragment back stacks are the stuff of nightmares (what's new?): the fragment isn't
+        // destroyed so its state isn't saved and we have to rely on an instance field. Sigh.
+        mCalled = true;
+
+        // Check for phone
+        // It is assumed that the phone number that are being wired in via Credential Selector
+        // are e164 since we store it.
+        Bundle params = getArguments().getBundle(ExtraConstants.PARAMS);
+        String phone = null;
+        String countryIso = null;
+        String nationalNumber = null;
+        if (params != null) {
+            phone = params.getString(ExtraConstants.PHONE);
+            countryIso = params.getString(ExtraConstants.COUNTRY_ISO);
+            nationalNumber = params.getString(ExtraConstants.NATIONAL_NUMBER);
+        }
+
+        if (!TextUtils.isEmpty(countryIso) && !TextUtils.isEmpty(nationalNumber)) {
```

Comment:
nit: This block is pretty much the heart of the decision making here so I think a comment or two would be good, especially for the final elseif block.